### ISSUE TITLE
What are the 'rust-analyzer diagnostics'?

### ISSUE THREAD

**Issue by lokegustafsson** [rust-analyzer/rust-analyzer#6214]
From the readme:
> `rust-analyzer.diagnostics.disabled`: List of rust-analyzer diagnostics to disable, default: `[]`

I would like to disable the "This operation is unsafe and requires an unsafe block" error diagnostic, since it gives false positives with wasm-bindgen. But I have no idea what to write in that array...

**Comment by bjorn3** [comment]
In this case it is `missing-unsafe`.

**Comment by bjorn3** [comment]
What are the false positives by the way? What code gives them?

**Comment by lokegustafsson** [comment]
Oh, i managed to post this to the wrong repo. I though i sent it posted it to https://github.com/fannheyward/coc-rust-analyzer ops. It is their readme that i refer to.

**Comment by flodiebold** [comment]
We should document our diagnostics (we could probably auto-generate this like we do for assists).

**Comment by bjorn3** [comment]
This is the right repo for this kind of configuration questions. It is rust-analyzer that implements them. coc-rust-analyzer just forwards them. You need to put

```jsonc
{
    "rust-analyzer.diagnostics.disabled": ["missing-unsafe"]
}
```

in `$HOME/.config/nvim/coc-settings.json` to disable the diagnostic for coc-rust-analyzer.

**Comment by flodiebold** [comment]
There's already a bug for the false positives, #5412.

**Comment by lokegustafsson** [comment]
`missing-unsafe` solves my problem, thank you. Regarding the false positive: wasm-bindgen is all about making declarations along the line of ```rust
#[wasm_bindgen]
extern "C" {
    #[wasm_bindgen(js_namespace = Console)]
    pub fn log(msg: &str);
}
```
This would make `log` unsafe as it is an external function, but some wasm-bindgen magic makes them safe. I don't know any details, but the code compiles and can use the function normally. However, rust-analyzer considers the function unsafe.

**Comment by lokegustafsson** [comment]
Yes, this is the same false positive. And yes, there should be a list of these diagnostics somewhere

**Comment by jplatte** [comment]
Found this: https://github.com/rust-analyzer/rust-analyzer/blob/c483212f274e9a5e348451cd4bbd0487e172458b/docs/user/generated_diagnostic.adoc

Is this also available somewhere on the website? Haven't found an obvious entry in the user manual TOC.

**Comment by flodiebold** [comment]
It should be included in the manual with the next release.
